dbms/src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -133,6 +133,15 @@ class IAggregateFunction
       */
     virtual void addBatch(size_t batch_size, AggregateDataPtr * places, size_t place_offset, const IColumn ** columns, Arena * arena) const = 0;
 
+    /** In addition to the above method, this variant accepts an array of "offsets" which allows
+      *  collecting multiple rows of arguments into array "places" as long as they are between
+      *  offsets[i-1] and offsets[i]. It is used for arrayReduce and might be used generally to
+      *  break data dependency when array "places" contains a large number of same values
+      *  consecutively.
+      */
+    virtual void addBatch(
```

Comment:
It's better to change the name. For example, addBatchArrays.
BTW, maybe, it can also be used to optimize -Array aggregate function combinator.